### PR TITLE
Fix avvio plugin timeout for 0

### PR DESCRIPTION
**Pull request by Miladiir**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by Miladiir** [comment]
I was looking for a better place to put the unit test, but couldn't find one. Anyway, here it is.

**Review by mcollina — APPROVED** [review]
lgtm

File: fastify.js [hunk 1]
```diff
@@ -352,9 +352,11 @@ function fastify (options) {
   // - ready
   // - onClose
   // - close
+
+  const avvioPluginTimeout = Number(options.pluginTimeout)
   const avvio = Avvio(fastify, {
     autostart: false,
-    timeout: Number(options.pluginTimeout) || defaultInitOptions.pluginTimeout,
+    timeout: !isNaN(avvioPluginTimeout) ? avvioPluginTimeout : defaultInitOptions.pluginTimeout,
```
Comment (jsumners) [inline fastify.js:359]:
```suggestion
    timeout: isNaN(avvioPluginTimeout) !== false ? avvioPluginTimeout : defaultInitOptions.pluginTimeout,
```

**Comment by mcollina** [comment]
Why https://github.com/fastify/fastify/blob/main/build/build-validation.js is not working for this parameter?

File: fastify.js (hunk 1 above)
Comment (Miladiir) [reply fastify.js:359]: `=== false` since this is a check if the variable is not NaN

**Comment by Miladiir** [comment]
I am not certain why it was implemented the way it was before, but the `|| defaultTimeout` part breaks my expectation that a `0` disables the timeout. The validation is still performed by ajv in build-validation. This check is only needed to check for 0, undefined or a valid timeout. Although funnily enough, negative numbers are considered valid.

**Comment by mcollina** [comment]
> I am not certain why it was implemented the way it was before, but the || defaultTimeout part breaks my expectation that a 0 disables the timeout. The validation is still performed by ajv in build-validation. This check is only needed to check for 0, undefined or a valid timeout. Although funnily enough, negative numbers are considered valid.

Why you expected 0 to disable the timeout?

**Comment by Miladiir** [comment]
https://github.com/fastify/help/issues/188
https://github.com/fastify/fastify/issues/3665

**Comment by mcollina** [comment]
Ah great, thanks!

**Comment by mcollina** [comment]
I think this is best as a semver-major change. Will be part of v4.

**Review by climba03003 — APPROVED** [review]
LGTM.

**Pull request merged**